src/test/java/com/cloudant/tests/ComplexKeySerializationTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain key copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import com.cloudant.client.api.views.Key;
+import com.cloudant.client.internal.views.ComplexKeyDeserializer;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import org.junit.Test;
+
+public class ComplexKeySerializationTest {
+
+    private static final Gson gson = new GsonBuilder().registerTypeAdapter(Key.ComplexKey.class,
+            new ComplexKeyDeserializer()).create();
+    public static final String STR_CMPLX_KEY_JSON = "[\"dog\",\"cat\",\"mouse\"]";
+    public static final String BOOL_CMPLX_KEY_JSON = "[true,false,true]";
+    public static final String NUM_CMPLX_KEY_JSON = "[1,12,15.0,99999999999999]";
+    public static final String MIXED_CMPLX_KEY_JSON = "[\"dog\",true,1,15.0,null]";
+
```

Comment:
Should we still keep the test for double quotes and spaces in strings? i.e. \" " test " \"